itest/mint_batch_stress_test.go

```diff
@@ -0,0 +1,185 @@
+
+	// Generate the rest of the batch, with each asset referencing the group
+	// anchor we created above.
+	for i := 1; i < batchSize; i++ {
+		groupedAsset := copyRequest(&collectibleRequestTemplate)
+		incrementMintAsset(groupedAsset.Asset, i)
+		groupedAsset.Asset.GroupAnchor = collectibleAnchorReq.Asset.Name
+		batchReqs[i] = groupedAsset
+	}
+
+	// Submit the batch for minting. Use an extended timeout for the TX
+	// appearing in the mempool so we can observe the minter hitting its own
+	// shorter default timeout.
+	t.LogfTimestamped("beginning minting of batch of %d assets", batchSize)
+	mintBatch := mintAssetsConfirmBatch(t, t.tapd, minterTimeout, batchReqs)
+	t.LogfTimestamped("finished batch mint of %d assets", batchSize)
+
+	// We can re-derive the group key to verify that the correct asset was
+	// used as the group anchor.
+	collectibleAnchor := verifyGroupAnchor(
+		t.t, mintBatch, collectibleAnchorReq.Asset.Name,
+	)
+	collectGroupKey := collectibleAnchor.AssetGroup.TweakedGroupKey
+	collectGroupKeyStr := hex.EncodeToString(collectGroupKey[:])
+
+	// We should have one group, with the specified number of assets and an
+	// equivalent balance, since the group is made of collectibles.
+	groupCount := 1
+	groupBalance := batchSize
+	assertNumGroups(t.t, t.tapd, groupCount)
+	assertGroupSizes(
+		t.t, t.tapd, []string{collectGroupKeyStr}, []int{batchSize},
+	)
+	assertBalanceByGroup(
+		t.t, t.tapd, collectGroupKeyStr, uint64(groupBalance),
+	)
+
+	// The universe tree should reflect the same properties about the batch;
+	// there should be one root with a group key and balance matching what
+	// we asserted previously.
+	ctx := context.Background()
+	uniRoots, err := t.tapd.AssetRoots(ctx, &unirpc.AssetRootRequest{})
+	require.NoError(t.t, err)
+	require.Len(t.t, uniRoots.UniverseRoots, groupCount)
+
+	err = assertUniverseRoot(
+		t.t, t.tapd, groupBalance, nil, collectGroupKey,
+	)
+	require.NoError(t.t, err)
+
+	// The universe tree should also have a leaf for each asset minted.
+	// TODO(jhb): Resolve issue of 33-byte group key handling.
+	collectUniID := unirpc.ID{
+		Id: &unirpc.ID_GroupKey{
+			GroupKey: collectGroupKey[1:],
+		},
+	}
+	uniLeaves, err := t.tapd.AssetLeaves(ctx, &collectUniID)
+	require.NoError(t.t, err)
+	require.Len(t.t, uniLeaves.Leaves, batchSize)
+
+	// The universe tree should also have a key for each asset, with all
+	// outpoints matching the chain anchor of the group anchor.
+	mintOutpoint := collectibleAnchor.ChainAnchor.AnchorOutpoint
+	uniKeys, err := t.tapd.AssetLeafKeys(ctx, &collectUniID)
+	require.NoError(t.t, err)
+	require.Len(t.t, uniKeys.AssetKeys, batchSize)
+
+	correctOp := fn.All(uniKeys.AssetKeys, func(key *unirpc.AssetKey) bool {
+		return key.GetOpStr() == mintOutpoint
+	})
+	require.True(t.t, correctOp)
+
+	// If we create a second tapd instance and sync the universe state,
+	// the synced tree should match the source tree.
+	bob := setupTapdHarness(
+		t.t, t, t.lndHarness.Bob, nil,
```

Comment:
Can't we replace the code that adds the federation server by just adding
```
		func(params *tapdHarnessParams) {
			params.startupSyncNode = t.tapd
			params.startupSyncNumAssets = len(rpcAssets)
		},
```
here?